dart/lib/src/client.dart

```diff
@@ -19,38 +17,13 @@ abstract class SentryClient {
   /// `dart:html` is available, otherwise it will throw an unsupported error.
   factory SentryClient(SentryOptions options) => createSentryClient(options);
 
-  SentryClient.base(
-    this.options, {
-    String platform,
-    this.origin,
-    Sdk sdk,
-  })  : _dsn = Dsn.parse(options.dsn),
-        _platform = platform ?? sdkPlatform,
-        sdk = sdk ?? Sdk(name: sdkName, version: sdkVersion);
-
-  final Dsn _dsn;
+  SentryClient.base(this.options, {@required this.transport});
```

Comment:
`options` already have a `transport` field, lets take from the `options` instead of passing as named param.